module/x/gravity/abci.go

```diff
@@ -371,33 +371,39 @@ func outgoingTxSlashing(ctx sdk.Context, k keeper.Keeper) {
 		}
 	}
 
+	slashedVals := make(map[string]bool)
```

Comment:
In EndBlocker, if a validator missed signatures for two outgoing txs, the validator could be slashed two times, which would panic the application

Reply:
interesting, I wonder how this could have not been detected.

Doesnt this line prevents the val to be slashed twice?

```
if !valInfo.val.IsJailed() {
```

Reply:
No, if this validator is slashed for the first time, the `valInfo` is not updated, so for another outgoingtx this validator could be slashed again for not signing the tx and this would panic the application.
Actually this is detected during simulation.

Reply:
or we could choose to update `valInfo` after the validator is jailed

Reply:
I see, I think either its fine

maybe you could replace 

```
if !valInfo.val.IsJailed()
```

by 

```
if !valInfo.val.IsJailed() && !slashedVals[valInfo.cons.String()]
```

It seems it has to be added in two location.

for ustxs and otx.(*types.SignerSetTx) loop within the same block

Reply:
yes, you are right.
replaced and added.

Reply:
I mean I think the same logic needs to be applied there as well?
https://github.com/crypto-org-chain/gravity-bridge/blob/67c72dc47870ffd58509de37c0ef257e15f8aee7/module/x/gravity/abci.go#L422

Could use the same map or another map